BackendApp/shelterRelated/tests.py

```diff
@@ -0,0 +1,122 @@
+        }
+        self.user = User.objects.create_user(
+            username=user_data["username"], password=user_data["password"]
+        )
+        url = reverse_lazy("login")
+        request = self.factory.post(url, user_data, format="json")
+        view = LoginView.as_view()
+        response = view(request)
+        self.token = response.data["access"]
+        shelter = Shelter.objects.create()
+        permission_create = Permission.objects.get(codename="add_shelter")
+        permission_change = Permission.objects.get(codename="change_shelter")
+        permission_delete = Permission.objects.get(codename="delete_shelter")
+        UserPermission.objects.create(
+            user=self.user, shelter=shelter, permission=permission_create
+        )
+        UserPermission.objects.create(
+            user=self.user, shelter=shelter, permission=permission_change
+        )
+        UserPermission.objects.create(
+            user=self.user, shelter=shelter, permission=permission_delete
+        )
+
+    def test_list(self):
+        url = reverse_lazy("shelter_list")
+        request = self.factory.get(url)
+        view = ShelterList.as_view()
+        response = view(request)
+        self.assertEqual(
+            response.status_code,
+            200,
+            f"Expected Response Code 200, received {response.status_code} instead.",
+        )
+
+    def test_create(self):
+        actual_shelter_count = Shelter.objects.count()
+        self.client.force_authenticate(user=self.user)
+        url = reverse_lazy("shelter_create")
+        response = self.client.post(url, self.data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Shelter.objects.count(), actual_shelter_count + 1)
+        self.assertEqual(
+            Shelter.objects.get(id=actual_shelter_count + 1).name, self.data["name"]
+        )
+
+    def test_detail(self):
+        Shelter.objects.create()
+        url = reverse_lazy("shelter_detail", kwargs={"pk": 1})
+        request = self.factory.get(url)
+        view = ShelterDetail.as_view()
+        response = view(request, pk=1)
+        self.assertEqual(
+            response.status_code,
+            200,
+            f"Expected Response Code 200, received {response.status_code} instead.",
+        )
+
+    def test_edit(self):
+        shelter = Shelter.objects.create()
+        url = reverse_lazy("shelter_detail", kwargs={"pk": 1})
+        data = self.data
+        data["name"] = "shelter123"
+        request = self.factory.put(url, data)
+        force_authenticate(request, user=self.user, token=self.token)
+        view = ShelterDetail.as_view()
+        response = view(request, pk=1)
+        self.assertEqual(
+            response.status_code,
+            200,
+            f"Expected Response Code 200, received {response.status_code} instead.",
+        )
+        self.assertEqual(Shelter.objects.get(id=1).name, data["name"])
```

Comment:
We should see the change here - assert that before the change name is not `data["name"]`